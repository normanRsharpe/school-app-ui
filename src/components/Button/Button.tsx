import React, { ReactNode } from 'react';

interface IButtonProps {
  disabled?: boolean;
  children?: ReactNode | string;
  name?: string;
  onClick?: (e: React.MouseEvent) => void;
  type?: 'button' | 'submit' | 'reset';
}

const Button: React.FC<IButtonProps> = ({
  children,
  disabled = false,
  name,
  onClick,
  type = 'button',
}) => (
  <button
    className="bg-moon-gray ph3 pv2 dim fw4 f3 dark-gray pointer br2 flex items-center center"
    disabled={disabled}
    name={name}
    onClick={onClick}
    type={type}
  >
    {children}
  </button>
);

export default Button;
