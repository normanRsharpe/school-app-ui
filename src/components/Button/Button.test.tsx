import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('<Button>', () => {
    it(':renders children', () => {
        const component = renderer.create(<Button>Test String</Button>).toJSON();
        expect(component).toMatchSnapshot();
    });
    it(':can have a border', () => {
        const component = renderer.create(
            <Button
                hasBorder={true}>
                Test String
            </Button>
        ).toJSON();
        expect(component).toMatchSnapshot();
    });
    it(':can be disabled', () => {
        const component = renderer.create(
            <Button
                disabled={true}>
                Test String
            </Button>
        ).toJSON();
        expect(component).toMatchSnapshot();
    });
});
