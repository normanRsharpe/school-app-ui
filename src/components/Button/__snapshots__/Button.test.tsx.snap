// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> :can be disabled 1`] = `
<button
  className="bn bg-transparent pa2 dim fw4 f6 gray ba b--moon-gray gray pointer flex items-center"
  disabled={true}
  type="button"
>
  Test String
</button>
`;

exports[`<Button> :can have a border 1`] = `
<button
  className="ba bg-transparent pa2 dim fw4 f6 gray ba b--moon-gray gray pointer flex items-center"
  disabled={false}
  type="button"
>
  Test String
</button>
`;

exports[`<Button> :renders children 1`] = `
<button
  className="bn bg-transparent pa2 dim fw4 f6 gray ba b--moon-gray gray pointer flex items-center"
  disabled={false}
  type="button"
>
  Test String
</button>
`;
